src/components/VideoPreview.tsx

```diff
@@ -1,36 +1,97 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useVideo } from '@/api/hooks'
 import routes from '@/config/routes'
-import VideoPreviewBase, { VideoPreviewBaseMetaProps } from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import VideoPreviewBase, {
+  VideoPreviewBaseMetaProps,
+  VideoPreviewPublisherProps,
+} from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import { useDrafts } from '@/hooks'
+import { copyToClipboard } from '@/utils/broswer'
 
-type VideoPreviewProps = {
+export type VideoPreviewProps = {
   id?: string
   progress?: number
+  isLoading?: boolean
   className?: string
 } & VideoPreviewBaseMetaProps
 
-const VideoPreview: React.FC<VideoPreviewProps> = ({ id, progress, className, ...metaProps }) => {
+const VideoPreview: React.FC<VideoPreviewProps> = ({ id, className, isLoading = false, ...metaProps }) => {
+  const [loadOnMount, setLoadOnMount] = useState(true)
   const { video, loading } = useVideo(id ?? '', { fetchPolicy: 'cache-first', skip: !id })
-  const isLoading = loading || id === undefined
+
+  useEffect(() => {
+    setLoadOnMount(false)
+  }, [])
+
+  const internalLoadingState = loading || !id || isLoading || loadOnMount
+  const videoHref = id ? routes.video(id) : undefined
   return (
     <VideoPreviewBase
+      publisherMode={false}
       title={video?.title}
       channelHandle={video?.channel.handle}
       channelAvatarUrl={video?.channel.avatarPhotoUrl}
       createdAt={video?.createdAt}
       duration={video?.duration}
       views={video?.views}
       thumbnailUrl={video?.thumbnailUrl}
-      videoHref={id ? routes.video(id) : undefined}
+      videoHref={videoHref}
       channelHref={id ? routes.channel(video?.channel.id) : undefined}
-      progress={progress}
-      isLoading={isLoading}
+      isLoading={internalLoadingState}
       className={className}
-      // TODO: handle menu contextMenuCallbacks
-      // contextMenuCallbacks={{}}
       {...metaProps}
     />
   )
 }
 
 export default VideoPreview
+
+export type VideoPreviewWPublisherProps = Omit<
+  {
+    id?: string
+    progress?: number
+    isLoading?: boolean
+    className?: string
+  } & VideoPreviewBaseMetaProps &
+    VideoPreviewPublisherProps,
+  'publisherMode' | 'videoPublishState'
+>
+export const VideoPreviewPublisher: React.FC<VideoPreviewWPublisherProps> = ({
+  id,
+  className,
+  isLoading = false,
+  isDraft,
+  ...metaProps
+}) => {
```

Comment:
@kdembler  Added the new component here but wasn't too sure about if it was the best way for the reusability of the logic